src/bun.js/api/server.zig

```diff
@@ -7192,6 +7246,20 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
                     );
                 },
             }
+
+            if (globalThis.hasException()) {
+                this.deinit();
+                return;
+            }
+
+            this.ref();
```

Comment:
what is the possible exception at this point?

Reply:
I"m not 100% sure ,but it felt worth checking

Reply:
onListenFailed